test/solver_tests/percolation_test.py

```diff
@@ -0,0 +1,200 @@
+"""
+Test PercolationSolver in Cassiopeia.solver.
+"""
+
+import unittest
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+from typing import Dict, List, Optional
+
+import cassiopeia as cas
+from cassiopeia.data import utilities as data_utilities
+from cassiopeia.solver.VanillaGreedySolver import VanillaGreedySolver
+from cassiopeia.solver.NeighborJoiningSolver import NeighborJoiningSolver
+from cassiopeia.solver.PercolationSolver import PercolationSolver
+from cassiopeia.solver import dissimilarity_functions
+
+
+def neg_hamming_similarity_without_missing(
+    s1: np.array,
+    s2: np.array,
+    missing_state_indicator: int,
+    weights: Optional[Dict[int, Dict[int, float]]] = None,
+) -> float:
+    """A function to return the number of (non-missing) character/state
+    mutations shared by two samples.
+
+    Args:
+        s1: Character states of the first sample
+        s2: Character states of the second sample
+        missing_state_indicator: The character representing missing values
+        weights: A set of optional weights to weight the similarity of a mutation
+    Returns:
+        The number of shared mutations between two samples, weighted or unweighted
+    """
+
+    # TODO Optimize this using masks
+    similarity = 0
+    for i in range(len(s1)):
+
+        if (
+            s1[i] == missing_state_indicator
+            or s2[i] == missing_state_indicator
+            or s1[i] == 0
+            or s2[i] == 0
+        ):
+            continue
+
+        if s1[i] == s2[i]:
+            if weights:
+                similarity += weights[i][s1[i]]
+            else:
+                similarity += 1
+
+    return -1 * similarity
```

Comment:
Is there a reason why this is not in `dissimilarity_functions`? This looks like it could be useful.

Reply:
This is just the negative of the hamming_similarity_without_missing, but numba doesn't like lambda functions otherwise I would've used a lambda here

Reply:
Okay, I see. Then to reduce the amount of code - why not use the following code block?

```
def neg_hamming_similarity_without_missing(...):
     return -1 * hamming_similarity_without_missing(...)
```

Reply:
Apparently numba doesn't like this either haha...

Reply:
I got it to work - looks like you have to compile the dissimilarity function with numba first!

Reading more into it, it looks like that if you'd like to call a function from a numba function, then you need to make sure that function within is compiled with numba.